File: Java/TestP4.java
package civitas;


public class TestP4{

    public static void main(String args[]) {
        Jugador jugador = new Jugador("Manolo");
        TituloPropiedad titulo = new TituloPropiedad("Finca Manoli", 400, 2, 12000, 15000, 300);
        jugador.getPropiedades().add(titulo);
        
        System.out.println("---Jugador antes de convertir:");
        System.out.println(jugador.toString());
        
        JugadorEspeculador especulador = new JugadorEspeculador(jugador, 500);
        System.out.println("\n---Jugador después de convertirse:");
        System.out.println(especulador.toString());
    }
}

File: Ruby/test_p4.rb
require_relative "jugador"
require_relative "jugador_especulador"
require_relative "titulo_propiedad"
module Civitas
  class TestP4
    def main
      
      jugador = Jugador.new("Manolo")
      titulo = TituloPropiedad.new("Finca Flores", 200, 2, 10000, 13000, 400)
      
      jugador.propiedades.push(titulo)
      puts "---Jugador antes de convertirse"
      puts jugador.to_s
      
      especulador = JugadorEspeculador.new(jugador, 2000)
     
      puts "---Jugador despues de convertirse"
      puts jugador.to_s
      
    end
  end
  
  test = TestP4.new
  test.main
end
